### PR TITLE
add binary and docs build command options to devmenu

### PR DESCRIPTION
**Pull request by aaronmarkham**
## Description ##
This PR will make it easier for users/contributors to build MXNet from source and then build any documentation set they need.

## Prerequisites
* Clone the repo
* Install Python3 as well as `pip3` (this script isn't compatible with Python 2.x)
* [Install Docker CE](https://docs.docker.com/engine/installation/linux/ubuntu/#install-using-the-repository)
* Install Docker for Python
```
pip3 install docker
```
* Get docker to run without sudo (this is for Ubuntu):
```bash
#!/bin/bash
set -e
set -x
export DEBIAN_FRONTEND=noninteractive
apt-get -y install curl
curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
add-apt-repository \
   "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
      $(lsb_release -cs) \
         stable"
apt-get update
apt-get -y install docker-ce
service docker restart
usermod -a -G docker $SUDO_USER
```

## Usage
Go to the root of the repo then run:
```
python3 devmenu.py
```

It'll look like this:
```

-- MXNet dev menu --

Available actions
------------------------------
1. [Local] BUILD CMake/Ninja (using cmake_options.yaml (cp cmake/cmake_options.yml .) and edit) (python3 virtualenv in "py3_venv")
2. [Local] Python Unit tests
3. [Docker] Build the MXNet binary - outputs to "lib/"
4. [Docker] Build the Jekyll website - outputs to "docs/static_site/build/html/"
5. [Docker] Build the Python API docs - outputs to "docs/python_docs/python/build/_build/html/"
6. [Docker] Build the C++ API docs - outputs to "docs/cpp_docs/build/html/html/"
7. [Docker] Build the Clojure API docs - outputs to "contrib/clojure-package/target/doc"
8. [Docker] Build the Java API docs - outputs to "docs/scala-package/build/docs/java"
9. [Docker] Build the Julia API docs - outputs to "julia/docs/site/"
10. [Docker] Build the R API docs - outputs to "R-package/build/mxnet-r-reference-manual.pdf"
11. [Docker] Build the Scala API docs - outputs to "scala-package/docs/build/docs/scala"
12. [Docker] sanity_check. Check for linting and code formatting and licenses.
13. [Docker] Python3 CPU unittests
14. [Docker] Python3 GPU unittests
15. [Docker] Python3 GPU+MKLDNN unittests
16. [Docker] Python3 CPU Intel MKLDNN unittests
17. [Docker] Python3 ARMv7 unittests (QEMU)
18. Clean (RESET HARD) repository (Warning! erases local changes / DATA LOSS)
------------------------------
Choose option> 
```

This PR adds options 3-11. 

With the exception of Jekyll, the other documentation sets are likely going to need you to build the MXNet binary first (running option `3`). But only once. When the binary is there, you can run the rest of the doc sets.

### CODE REVIEW

**Comment by aaronmarkham** [comment]
@larroy Can you review?

File: dev_menu.py [hunk 1]
```diff
@@ -132,8 +132,24 @@ def provision_virtualenv(venv_path=DEFAULT_PYENV):
     ('[Local] Python Unit tests',
         "./py3_venv/bin/nosetests -v tests/python/unittest/"
     ),
-    ('[Docker] Website and docs build outputs to "docs/_build/html/"',
-        "ci/build.py --platform ubuntu_cpu /work/runtime_functions.sh deploy_docs"),
+    ('[Docker] Build the MXNet binary - outputs to "lib/"',
```
Comment (larroy) [inline dev_menu.py]: mxnet binary? not clear what this does.

Comment (aaronmarkham) [reply dev_menu.py]: It builds the .so and .a file that the docs need for API docs generation.

Comment (aaronmarkham) [reply dev_menu.py]: only the .so file is needed AFAIK

**Pull request merged**